### PR TITLE
Distributed Tracing: Refactors code to rename net.peer.name attribute to server.address

### LINKED ISSUE #4309: OpenTelemetry net.peer.name attribute is out of date
The net.peer.name attribute [here](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/3f421fbb03f7f93950bd6eb16032ce4d852420c3/Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryAttributeKeys.cs#L19) is out of date.

address.server is the modern name. See https://github.com/open-telemetry/opentelemetry-specification/issues/3199

Should also add peer.service which is a general name: https://github.com/open-telemetry/semantic-conventions/blob/482cb01c4598429c65dd0b123b30259f41b360cb/docs/general/attributes.md#general-remote-service-attributes

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description
Renaming `net.peer.name` to `server.address` ref. #4309

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #4309

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs [hunk 1]
```diff
@@ -22,7 +22,8 @@ internal static class OpenTelemetryRecorderFactory
                            clientNamespace: $"{OpenTelemetryAttributeKeys.DiagnosticNamespace}",
                            resourceProviderNamespace: OpenTelemetryAttributeKeys.ResourceProviderNamespace,
                            isActivityEnabled: true,
-                           suppressNestedClientActivities: true),
+                           suppressNestedClientActivities: true, 
+                           isStable: true),
```
Comment (jcocchi) [inline Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs:26]: Are we marking all of our attributes as stable? Or is this related to removing the `Azure.Experimental.EnableActivitySource` tab?

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs:26]: It will remove the env variable to enable the feature...for attribute name, one has to refer the otel doc.

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs (hunk 1 above)
Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs]: @jcocchi  as we discussed about stability part..I have removed that change from this PR. Now `Experimental` env variable will be required

**Review by kirankumarkolli — DISMISSED** [review]

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs [hunk 2]
```diff
@@ -22,7 +22,8 @@ internal static class OpenTelemetryRecorderFactory
                            clientNamespace: $"{OpenTelemetryAttributeKeys.DiagnosticNamespace}",
                            resourceProviderNamespace: OpenTelemetryAttributeKeys.ResourceProviderNamespace,
                            isActivityEnabled: true,
-                           suppressNestedClientActivities: true),
+                           suppressNestedClientActivities: true, 
+                           isStable: false),
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs:26]: nit: Is this change related to the rename?

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs (hunk 2 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryRecorderFactory.cs:26]: It was removed a bit too early in a previous PR, we want to wait for the OTel semantic conventions to reach stability before marking our instrumentation as stable

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/EndToEndTraceWriterBaselineTests.BatchOperationsAsync.xml [hunk 3]
```diff
@@ -147,13 +147,13 @@
...
     <ATTRIBUTE key="db.cosmosdb.machine_id">Some Value</ATTRIBUTE>
-    <ATTRIBUTE key="net.peer.name">127.0.0.1</ATTRIBUTE>
+    <ATTRIBUTE key="server.address">127.0.0.1</ATTRIBUTE>
     <ATTRIBUTE key="db.cosmosdb.client_id">Some Value</ATTRIBUTE>
     <ATTRIBUTE key="user_agent.original">Some Value</ATTRIBUTE>
     <ATTRIBUTE key="db.cosmosdb.connection_mode">Direct</ATTRIBUTE>
     <ATTRIBUTE key="db.cosmosdb.operation_type">Batch</ATTRIBUTE>
-    <ATTRIBUTE key="db.cosmosdb.status_code">400</ATTRIBUTE>
-    <ATTRIBUTE key="db.cosmosdb.sub_status_code">1001</ATTRIBUTE>
+    <ATTRIBUTE key="db.cosmosdb.status_code">Some Value</ATTRIBUTE>
```
Comment (jcocchi) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/EndToEndTraceWriterBaselineTests.BatchOperationsAsync.xml:155]: Is it intentional these aren't "real" status codes anymore?

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/EndToEndTraceWriterBaselineTests.BatchOperationsAsync.xml:155]: only for tests..I am not putting these in XMLs because order of these events are changing and making tests flaky...nothing is changed here (except in test)
